Data/models/Bookmark.swift

```diff
@@ -379,6 +379,16 @@ public final class Bookmark: NSManagedObject, WebsitePresentable, Syncable, CRUD
         delete()
     }
     
+    /// Removes a single Bookmark of a given URL.
+    /// In case of having two bookmarks with the same url, a bookmark to delete is chosen randomly.
+    public class func remove(forUrl url: URL) {
+        let context = DataController.newBackgroundContext()
+        let predicate = isFavoriteOrBookmarkByUrlPredicate(url: url, getFavorites: false)
+        
+        let record = first(where: predicate, context: context)
+        record?.remove()
```

Comment:
The only difference is using `remove()` instead of `delete()`

It's confusing to have those two methods. I will try to fix this in another PR, to make `Syncable` CRUD methods easier to distinguish from regular `CRUDProtocol` methods.